libtool/run.tcc

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2015-2021 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#ifndef __ZETH_LIBTOOL_RUN_TCC__
+#define __ZETH_LIBTOOL_RUN_TCC__
+
+#include "libtool/run.hpp"
+
+namespace libtool
+{
+
+namespace internal
+{
+
+template<typename GlobalOptionsT>
+void print_usage(
+    const char *const *argv,
+    const po::options_description &options,
+    const std::map<std::string, subcommand<GlobalOptionsT> *> &subcommands)
+{
+    std::cout << "Usage:\n"
+              << "  " << argv[0]
+              << " [<options>] <command> <command-arguments> ...\n\n"
+              << options;
+
+    std::cout << "\nCommands:\n";
+
+    // list_commands(commands);
+    using entry_t = std::pair<std::string, subcommand<GlobalOptionsT> *>;
+
+    // +4 here to ensure minimal space between command name and description
+    const size_t cmd_name_padded =
+        4 + std::max_element(
+                subcommands.begin(),
+                subcommands.end(),
+                [](const entry_t &a, const entry_t &b) {
+                    return a.first.size() < b.first.size();
+                })
+                ->first.size();
+
+    for (const auto &cmd : subcommands) {
+        const size_t padding = cmd_name_padded - cmd.first.size();
+        std::cout << "  " << cmd.first << std::string(padding, ' ')
+                  << cmd.second->description() << "\n";
+    }
+
+    std::cout << std::endl;
+}
+
+template<typename GlobalOptionsT>
+int run_subcommand(
+    const std::map<std::string, subcommand<GlobalOptionsT> *> &subcommands,
+    const std::string &command_name,
+    const char *argv0,
+    const std::vector<std::string> &command_args,
+    const GlobalOptionsT &global_options)
+{
+    subcommand<GlobalOptionsT> *sub = subcommands.find(command_name)->second;
+    if (sub == nullptr) {
+        throw po::error("invalid command");
+    }
+
+    return sub->execute(argv0, command_args, global_options);
+}
+
+} // namespace internal
+
+template<typename GlobalOptionsT>
+int run_command(
+    command<GlobalOptionsT> &command,
+    GlobalOptionsT &options,
+    const std::map<std::string, subcommand<GlobalOptionsT> *> &subcommands,
+    int argc,
+    char **argv)
+{
+    po::options_description global("Global options");
+    po::options_description all("");
+
+    // Default --help option
+    global.add_options()("help,h", "This help");
```

Comment:
`"This help"` -> `"Show this message and exit."`

Reply:
Change flag description text